adareducer/engine.py

```diff
@@ -331,6 +316,37 @@ def reduce_file(self, file):
             buf = Buffer(file)
             chars_removed = count - buf.count_chars()
 
+        return chars_removed
+
+    def reduce_file(self, file):
+        """Reduce one given file as much as possible"""
+
+        self.mark_as_processed(file)
```

Comment:
I wonder if we should mark as processed after actually processing it?

Reply:
Not a very strong opinion, but I think the body of this function is slightly more maintainable this way - setting this flag further below means we need to set this systematically before each "return" or "raise"... or add it in a global try/finally block which is a bit awkward.

Reply:
Good point.